core/services/chainlink/config.go

```diff
@@ -4,36 +4,66 @@ import (
 	"strings"
 
 	"github.com/pelletier/go-toml/v2"
+	"go.uber.org/multierr"
 
 	solcfg "github.com/smartcontractkit/chainlink-solana/pkg/solana/config"
 	soldb "github.com/smartcontractkit/chainlink-solana/pkg/solana/db"
 	tercfg "github.com/smartcontractkit/chainlink-terra/pkg/terra/config"
 	terdb "github.com/smartcontractkit/chainlink-terra/pkg/terra/db"
-
 	evmcfg "github.com/smartcontractkit/chainlink/core/chains/evm/config/v2"
 	evmtyp "github.com/smartcontractkit/chainlink/core/chains/evm/types"
 	"github.com/smartcontractkit/chainlink/core/chains/solana"
 	tertyp "github.com/smartcontractkit/chainlink/core/chains/terra/types"
+	coreconfig "github.com/smartcontractkit/chainlink/core/config"
 	config "github.com/smartcontractkit/chainlink/core/config/v2"
+	"github.com/smartcontractkit/chainlink/core/logger"
 	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
 // Config is the root type used for TOML configuration.
 //
+// See docs at /docs/CONFIG.md generated via config.GenerateDocs from /internal/config/docs.toml
+//
 // When adding a new field:
-// 	- consider including a unit suffix with the field name
-// 	- TOML is limited to int64/float64, so fields requiring greater range/precision must use non-standard types
-//	implementing encoding.TextMarshaler/TextUnmarshaler, like utils.Big and decimal.Decimal
-//  - std lib types that don't implement encoding.TextMarshaler/TextUnmarshaler (time.Duration, url.URL, big.Int) won't
-//   work as expected, and require wrapper types. See models.Duration, models.URL, utils.Big.
+//   - consider including a unit suffix with the field name
+//   - TOML is limited to int64/float64, so fields requiring greater range/precision must use non-standard types
+//     implementing encoding.TextMarshaler/TextUnmarshaler, like utils.Big and decimal.Decimal
+//   - std lib types that don't implement encoding.TextMarshaler/TextUnmarshaler (time.Duration, url.URL, big.Int) won't
+//     work as expected, and require wrapper types. See models.Duration, models.URL, utils.Big.
 type Config struct {
 	config.Core
 
-	EVM []EVMConfig `toml:",omitempty"`
+	EVM EVMConfigs `toml:",omitempty"`
 
-	Solana []SolanaConfig `toml:",omitempty"`
+	Solana SolanaConfigs `toml:",omitempty"`
 
-	Terra []TerraConfig `toml:",omitempty"`
+	Terra TerraConfigs `toml:",omitempty"`
+}
+
+func NewConfig(tomlString string, lggr logger.Logger) (coreconfig.GeneralConfig, error) {
```

Comment:
Does this function need to also take TOMLs for secrets and runtime config? (to pass them to the legacy wrapper)

Reply:
Yeah, secerets here too probably